fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/impl/networking/AbstractChanneledNetworkAddon.java

```diff
@@ -87,6 +90,10 @@ public boolean handle(CustomPayload payload) {
 			}
 		}
 
+		if (payload instanceof BrandCustomPayload brand) {
+			this.brand = brand.brand();
```

Comment:
I'd return false after this point. False becuase we only need to "peek" at this packet and not consume it, allowing vanilla or any exisitng mod to also read it.